ddl/ddl_api.go

```diff
@@ -4305,57 +4305,97 @@ func (d *ddl) TruncateTable(ctx sessionctx.Context, ti ast.Ident) error {
 
 func (d *ddl) RenameTable(ctx sessionctx.Context, oldIdent, newIdent ast.Ident, isAlterTable bool) error {
 	is := d.GetInfoSchemaWithInterceptor(ctx)
+	if is.TableExists(newIdent.Schema, newIdent.Name) {
+		return infoschema.ErrTableExists.GenWithStackByArgs(newIdent)
+	}
```

Comment:
It can be moved into `extractTblInfos`, to keep it the same with the original code.